### PR TITLE
refactor: ScreenSizeController to use @rhds/tokens

### PR DESCRIPTION
**Pull request by zeroedin**
## What I did

1. Refactored `ScreenSizeController` to use token values from `@rhds/tokens`
2. Updated `rh-navigation-secondary`, `rh-footer`, and `rh-stat` to use the new token keys.
3. Updated `rh-footer-responsive.css` made the values static.  

BLOCKED by https://github.com/patternfly/patternfly-elements/pull/2462

## Testing Instructions

1. View the [deploy preview](https://deploy-preview-875--red-hat-design-system.netlify.app/), make sure to check footer, secondary-nav and stat.
2. [navigation-secondary demo](https://deploy-preview-875--red-hat-design-system.netlify.app/elements/navigation-secondary/demo/)

## Notes to Reviewers

I think the `rh-footer-responsive.css` file could just go into a standard CSS file moving forward rather than being represented as a typescript file.  That said I added comments above each media query with the prior token and new token and if the values sync or not.  If there is a desire to add the interpolation back, feel free to make that commit.

### CODE REVIEW

**Comment by zeroedin** [comment]
@bennypowers had an issue with the `@rhds/token/js/media.js` import locally, and it definitely breaks the DP here.   PTAL and feel free to commit away.  I definitely found it weird to be asking for a declaration for the module, assuming maybe something more needs to happen on the token side of things.

File: lib/ScreenSizeController.ts [hunk 1]
```diff
@@ -1,40 +1,59 @@
...
+  Media2xl,
+  MediaLg,
+  MediaMd,
+  MediaSm,
+  MediaXl,
+  MediaXs,
+} from '@rhds/tokens/media.js';
+
+type BreakpointKey =
+  | '2xs'
```
Comment (zeroedin) [inline lib/ScreenSizeController.ts:14]: Should this key be `2xsMax` as it references a `max-width` value `Breakpoint2xsMax`.  If we added the `2xs` min-width value later?

```suggestion
  | '2xsMax'
```

File: lib/ScreenSizeController.ts [hunk 2]
```diff
@@ -1,40 +1,59 @@
...
+
+function getMediaQueryListForToken(token: MediaToken | string) {
+  const media =
+      typeof token === 'string' ? `(max-width: ${token})`
+    : Object.entries(token).map(x => `(${x.join(':')})`).join(' and ');
+  return matchMedia(`screen and ${media}`);
+}
+
+const BREAKPOINTS: Record<BreakpointKey, string | MediaToken> = {
+  '2xs': Breakpoint2xsMax,
```
Comment (zeroedin) [inline lib/ScreenSizeController.ts:38]:
```Suggestion
  '2xsMax': Breakpoint2xsMax,
```

File: lib/ScreenSizeController.ts [hunk 3]
```diff
@@ -1,40 +1,59 @@
...
-    ['desktopLarge', matchMedia(`screen and (min-width: ${desktopLargeBreakpoint})`)],
-  ]);
+  static queries = new Map<
+    BreakpointKey,
+    MediaQueryList
+  >(Object.entries(BREAKPOINTS).map(([k, v]) =>
+    [k as BreakpointKey, getMediaQueryListForToken(v)]));
 
-  public mobile = ScreenSizeController.queries.get('mobile')?.matches ?? false;
+  public mobile = ScreenSizeController.queries.get('2xs')?.matches ?? false;
```
Comment (zeroedin) [inline lib/ScreenSizeController.ts:56]:
```Suggestion
  public mobile = ScreenSizeController.queries.get('2xsMax')?.matches ?? false;
```

File: lib/ScreenSizeController.ts [hunk 4]
```diff
@@ -51,12 +70,12 @@ export class ScreenSizeController implements ReactiveController {
...
   ) {
     this.host.addController(this);
-    this.size = 'mobilePortrait';
+    this.size = '2xs';
     this.breakpoint = breakpoint;
     this.onChange = options?.onChange;
 
     for (const [key, list] of ScreenSizeController.queries) {
-      if (key !== 'mobile' && list.matches) {
+      if (key !== '2xs' && list.matches) {
```
Comment (zeroedin) [inline lib/ScreenSizeController.ts:78]:
```Suggestion
      if (key !== '2xsMax' && list.matches) {
```

File: lib/ScreenSizeController.ts [hunk 5]
```diff
@@ -73,16 +92,16 @@ export class ScreenSizeController implements ReactiveController {
...
-      this.onChange?.(this.matches.has(this.breakpoint) ?? false);
+      this.onChange?.(this.matches.has(this.breakpoint));
     }
+    this.host.requestUpdate();
   }
 }
 
 for (const [key, list] of ScreenSizeController.queries) {
-  if (key === 'mobile') {
+  if (key === '2xs') {
```
Comment (zeroedin) [inline lib/ScreenSizeController.ts:104]:
```Suggestion
  if (key === '2xsMax') {
```

File: elements/rh-navigation-secondary/rh-navigation-secondary.ts [hunk 6]
```diff
@@ -1,31 +1,37 @@
...
 import { property } from 'lit/decorators/property.js';
-import { query } from 'lit/decorators/query.js';
-import { state } from 'lit/decorators/state.js';
 import { classMap } from 'lit/directives/class-map.js';
-import { bound, observed } from '@patternfly/pfe-core/decorators.js';
 import { Logger } from '@patternfly/pfe-core/controllers/logger.js';
 
 import '../rh-context-provider/rh-context-provider.js';
 
 import './rh-navigation-secondary-menu-section.js';
```
Comment (zeroedin) [inline elements/rh-navigation-secondary/rh-navigation-secondary.ts:10]: Overlay isn't loading, add import

```Suggestion
import './rh-navigation-secondary-overlay.js';

```

File: elements/rh-navigation-secondary/rh-navigation-secondary.ts [hunk 7]
```diff
@@ -41,83 +47,53 @@ import styles from './rh-navigation-secondary.css';
...
+   *  - If viewport is mobile, open mobile menu
+   *  - otherwise, close mobile menu and close overlay
    */
-  protected screenSize = new ScreenSizeController(this, 'tabletLandscape', {
+  #screenSize = new ScreenSizeController(this, 'md', {
     onChange: matches => {
-      this._compact = !matches;
+      const dropdownsOpen = this.#allDropdowns().some(x => x.expanded);
+      this.mobileMenuExpanded = matches && dropdownsOpen;
+      this.overlayOpen = matches && dropdownsOpen;
```
Comment (zeroedin) [inline elements/rh-navigation-secondary/rh-navigation-secondary.ts:93]: Should be when `!matches`

```Suggestion
      const dropdownsOpen = this.#allDropdowns().some(x => x.expanded);
      this.mobileMenuExpanded = !matches && dropdownsOpen;
      this.overlayOpen = !matches && dropdownsOpen;
```

File: lib/ScreenSizeController.ts (hunk 1 above)
Comment (bennypowers) [reply lib/ScreenSizeController.ts:14]: Let's keep it as is for now. It's the edge case, and if the tokens change we'll have to redo this anyways

**Comment by brianferry** [comment]
Stat is lookin good! 👍

**Review by bennypowers — APPROVED** [review]
Laws give them meaning

**Pull request merged**